introduce a cmdctrl  k command called report bug

filiphric opened the pull request:
Closes https://github.com/metabase/metabase/issues/49585 and https://github.com/metabase/metabase/issues/49584

### Description

Add Diagnostic data window to command palette and user settings menu

### How to verify

Describe the steps to verify that the changes are working as expected.

Search functionality
1. cmd + k
2. search for "report a bug".  also can by found by searching  words like: bug, report, issue, problem, error, diagnostic

User menu
1. open user menu
2. click on "report a bug" item

### Demo

![Screenshot 2024-11-13 at 13 19 04](https://github.com/user-attachments/assets/d2623712-c623-40b1-aa15-8efa819a4557)
![Screenshot 2024-11-13 at 13 18 52](https://github.com/user-attachments/assets/5f5c80a1-0f2b-4ac7-aa6e-f4ccc6d7897c)

mazameli reviewed (changes requested):
I asked @romnitt just now in Figma if we could use a better icon for the report-a-bug action rather than the `info` one.

(Separately, more of a comment or small worry than a change request: I wonder if @thebiglabasky [what a good GH handle] is concerned about people seeing this action too often when they're just searching for a dashboard or saved question that has the name `Report` in it. Will that give people too much of an impression that Metabase is so buggy that that action needs to pop up frequently?)

filiphric committed:
add bug report to command palette and profile menu

rafpaf reviewed (changes requested):
The interface and code look spiffing! Could you please add e2e tests for finding "Report a bug" in both places?

filiphric commented:
> The interface and code look spiffing! Could you please add e2e tests for finding "Report a bug" in both places?

wouldn’t unit test make more sense here?

filiphric commented:
I updated the icon and added a test to cover the scenario. I opted in for the e2e test, since I could not figure out an effective way of how to write a unit test 😅

rafpaf commented:
> > The interface and code look spiffing! Could you please add e2e tests for finding "Report a bug" in both places?
> 
> wouldn’t unit test make more sense here?

Yes, good call - unit tests would be fine and arguably even better.

thebiglabasky commented:
> (Separately, more of a comment or small worry than a change request: I wonder if @thebiglabasky [what a good GH handle] is concerned about people seeing this action too often when they're just searching for a dashboard or saved question that has the name `Report` in it. Will that give people too much of an impression that Metabase is so buggy that that action needs to pop up frequently?)

Right I hadn't thought of that. I wonder if we could trick that by using "File a bug" instead 😅?

filiphric commented:
> > (Separately, more of a comment or small worry than a change request: I wonder if @thebiglabasky [what a good GH handle] is concerned about people seeing this action too often when they're just searching for a dashboard or saved question that has the name `Report` in it. Will that give people too much of an impression that Metabase is so buggy that that action needs to pop up frequently?)
> 
> Right I hadn't thought of that. I wonder if we could trick that by using "File a bug" instead 😅?

I changed the text from "report a bug" to "file a bug" and removed "report" as a keyword so that it would not appear in the search too often. should I change this in the profile link menu as well?

thebiglabasky commented:
> I changed the text from "report a bug" to "file a bug" and removed "report" as a keyword so that it would not appear in the search too often. should I change this in the profile link menu as well?

Nice, I'd say so indeed to stay consistent

filiphric commented:
I updated this. it’s ready to merge once I get an approval

thebiglabasky commented:
There are quite some people here in the review list, maybe it's easier if we had one specific person? Also noting that this is Thanksgiving so unlikely for north american colleagues to jump in until Monday. Would anyone elsewhere be available?

uladzimirdev reviewed (approved):
Code itself LGTM, nice work.

As per product/design perspective, I'd double check if we need to show actions before the search results. I doubt users are looking for a "report bug" action often.

UPD: I see @mazameli had the same concern

filiphric commented:
@rafpaf @mazameli I implemented the changes can you re-review please?

npfitz reviewed (changes requested):
I believe you can already open that modal through the command palette via "Open error diagnostic modal".

Can we just change the action name?

filiphric commented:
> I believe you can already open that modal through the command palette via "Open error diagnostic modal".
> 
> Can we just change the action name?

you’re right, I did not know about this. I pushed changes to fix this.

npfitz commented on e2e/test/scenarios/onboarding/command-palette.cy.spec.js:
```diff
@@ -247,4 +247,14 @@ describe("command palette", () => {
     });
   });
+
+  it("should show the 'File a bug' command palette item", () => {
```
The functionality here would already be covered in https://github.com/metabase/metabase/blob/49585-introduce-a-cmdctrl--k-command-called-report-bug/e2e/test/scenarios/admin-2/error-reporting.cy.spec.ts#L48. Can we update this test to reflect the new name?

npfitz reviewed (changes requested):
As mentioned, I think there is an existing e2e test that can be adjusted for the Command Palette action. Can we also get some coverage on the new item in the drop down menu here?
![image](https://github.com/user-attachments/assets/50c40f4d-d2a2-4c7d-b084-93ca67b59c7d)

npfitz commented on frontend/src/metabase/nav/components/ProfileLink/ProfileLink.jsx:
```diff
@@ -157,6 +164,10 @@ function ProfileLink({ adminItems, onLogout }) {
       ) : null}
+
+      {modalOpen === "diagnostic" && (
+        <ErrorDiagnosticModalWrapper isModalOpen={true} onClose={closeModal} />
```
Something else is that this modal already exists in the React Component tree in `App.tsx` as `<KeyboardTriggeredErrorModal />`. Rather than have it here twice, can the "Filke a bug" action just dispatch `openDiagnostics()` as well? Rather than have the modal in the component tree twice?

mazameli reviewed (commented):
Changes seem good to me — the only weirdness is that you select an action that says "File a bug," and then you see a modal that says "download diagnostic info." I know that the plan is that we're going to update that modal so that it'll make sense with this new action — has the modal already been updated in `master` but just not in this branch?

<img width="647" alt="image" src="https://github.com/user-attachments/assets/56a9348b-f32a-429d-acf3-46ffa190bb48">

filiphric commented:
> Changes seem good to me — the only weirdness is that you select an action that says "File a bug," and then you see a modal that says "download diagnostic info." I know that the plan is that we're going to update that modal so that it'll make sense with this new action — has the modal already been updated in `master` but just not in this branch?
> 
> <img alt="image" width="647" src="https://private-user-images.githubusercontent.com/2223916/391769289-56a9348b-f32a-429d-acf3-46ffa190bb48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyMTQyMTMsIm5iZiI6MTczMzIxMzkxMywicGF0aCI6Ii8yMjIzOTE2LzM5MTc2OTI4OS01NmE5MzQ4Yi1mMzJhLTQyOWQtYWNmMy00NmZmYTE5MGJiNDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDNUMDgxODMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTNkZDA1ZDA0YTQyN2JmNGFmZjY1YzhkOGU4NmM4ZGUxMzNhNzFjM2ViMzg0ZTFiZWY4ZjU1NDQ5MjFhOTBkMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.firYFn61B7knqPFwXQ6JRe50oGeL5ZGXFaOpma_gAVU">

yes there’s a slight discrepancy in the wording. the good news is that we are slowly moving away from simply downloading information towards actually reporting a bug. this is the PR that is coming right after this one: https://github.com/metabase/metabase/pull/50277 and there’s a [whole project](https://www.notion.so/metabase/Quality-initiatives-update-14d69354c90180648b0fce2c0268e856?pvs=4) that will improve this flow. the "File a bug" will make much more sense in upcoming weeks.

filiphric committed:
Merge branch 'master' into 49585-introduce-a-cmdctrl--k-command-called-report-bug

mazameli commented:
@filiphric Not to be pedantic, but then shouldn't https://github.com/metabase/metabase/pull/50277 target _this_ branch, and then once it's integrated, we merge the whole together into `master`? Typically when something is merged to `master` it means we think it's complete. cc @thebiglabasky

filiphric commented:
> @filiphric Not to be pedantic, but then shouldn't #50277 target _this_ branch, and then once it's integrated, we merge the whole together into `master`? Typically when something is merged to `master` it means we think it's complete. cc @thebiglabasky

It’s okay to be pedantic, I’m still trying to grasp how things work around here. my approach was to work granularly and finish small tasks to push the big thing out in small chunks. I’m realizing this was probably not the best approach, since it takes a little while to push things into production. I’ll definitely change my style for the future, because right now I created a train of PRs piggybacking on top of each other and the whole experience was a pain. my PRs are stuck in review for 4 weeks, I’m pinging devs for reviews and switching contexts all day. I have 5 PRs now, all seem to be green so hopefully I can push them out of the way.

filiphric commented:
closed in favor of https://github.com/metabase/metabase/pull/50856
